inc/planningexternalevent.class.php

```diff
@@ -0,0 +1,163 @@
+      $this->showFormHeader($options);
+
+      echo "<tr class='tab_bg_2'><td colspan='2'>".__('Title')."</td>";
+      echo "<td colspan='2'>";
+      if (!$ID) {
+         echo Html::hidden('users_id', ['value' => $this->fields['users_id']]);
+      }
+      if ($canedit) {
+         Html::autocompletionTextField($this, "name", [
+            'size'   => '80',
+            'entity' => -1,
+            'user'   => $this->fields["users_id"],
+         ]);
+      } else {
+         echo $this->fields['name'];
+      }
+      if (isset($options['from_planning_edit_ajax']) && $options['from_planning_edit_ajax']) {
+         echo Html::hidden('from_planning_edit_ajax');
+      }
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_2'>";
+      echo "<td colspan='2'>".__('Status')."</td>";
+      echo "<td colspan='2'>";
+      if ($canedit) {
+         Planning::dropdownState("state", $this->fields["state"]);
+      } else {
+         echo Planning::getState($this->fields["state"]);
+      }
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_2'>";
+      echo "<tr class='tab_bg_2'>";
+      echo "<td colspan='2'>".__('Category')."</td>";
+      echo "<td colspan='2'>";
+      if ($canedit) {
+         PlanningEventCategory::dropdown([
+            'value' => $this->fields['planningeventcategories_id']
+         ]);
+      } else {
+         echo Dropdown::getDropdownName(
+            PlanningEventCategory::getTable(),
+            $this->fields['planningeventcategories_id']
+         );
+      }
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_2'>";
+      echo "<td colspan='2'>".__('Background event')."</td>";
+      echo "<td colspan='2'>";
+      if ($canedit) {
+         Dropdown::showYesNo('background', $this->fields['background']);
+      } else {
+         echo Dropdown::getYesNo($this->fields['background']);
+      }
+      echo "</td>";
+      echo "</tr>";
+
+      echo "<tr class='tab_bg_2'><td  colspan='2'>".__('Calendar')."</td>";
+      echo "<td>";
+      if (isset($options['from_planning_ajax'])
+          && $options['from_planning_ajax']) {
+         echo Html::hidden('plan[begin]', ['value' => $options['begin']]);
+         echo Html::hidden('plan[end]', ['value' => $options['end']]);
+         printf(__('From %1$s to %2$s'), Html::convDateTime($options["begin"]),
+                                         Html::convDateTime($options["end"]));
+      } else {
+         Planning::showAddEventClassicForm([
+            'items_id'  => $this->fields['id'],
+            'itemtype'  => $this->getType(),
+            'begin'     => $this->fields['begin'],
+            'end'       => $this->fields['end'],
+            'rand_user' => $this->fields['users_id'],
+         ]);
+      }
+      echo "</td></tr>";
+
+      echo "<tr class='tab_bg_2'><td  colspan='2'>".__('Repeat')."</td>";
+      echo "<td>";
+      echo self::showRepetitionForm($this->fields['rrule']);
```

Comment:
If field is null in DB, passed value is not a string so it triggers a type error.
```suggestion
      echo self::showRepetitionForm($this->fields['rrule'] ?? '{}');
```